test/main/assert.js

```diff
@@ -182,6 +186,17 @@ QUnit.test( "throws", function( assert ) {
 		"simple string throw, no 'expected' value given"
 	);
 
+	assert.throws( function() {
+		// eslint-disable-next-line qunit/no-throws-string
+		assert.throws(
+			function() {
+				throw "my error";
+			},
+			"expected",
+			"throws fail when expected is string and message is non-null"
```

Comment:
Is this meant to test throwing of a string, or the unsupported matching against an expected string? I suspect the latter, in which case we can use a throw an Error object instead.

If I understand correctly, the message is currently lost -  t should be on the outer `throws()`, right? Could you also add an expectation match for that one, e.g. `/does not accept a string value/`.

Reply:
I agree, this nested throw should be clarified. It's intent is to verify an exception is thrown when "expected" is a string and "message' is non-null. This is the relevant branches in src/assert.js that this covers:

```
// 'expected' is optional unless doing string comparison
if ( objectType( expected ) === "string" ) {
	if ( message == null ) {
		message = expected;
		expected = null;
	} else {
		throw new Error(
			"throws/raises does not accept a string value for the expected argument.\n" +
			"Use a non-string object value (e.g. regExp) instead if it's necessary."
		);
	}
}
```